docs/pages/advanced-topics/docker.md

```diff
@@ -0,0 +1,19 @@
+Main [Dockerfile](./Dockerfile) image is a [multi-platform](https://docs.docker.com/build/building/multi-platform/) and [multi-stage](https://docs.docker.com/build/building/multi-stage/) image. It builds on [node-alpine](https://hub.docker.com/_/node) to optimise for final size.
+
+A typical build command-line for creating an image for linux (`amd64` and `arm64`) would be:
+
+```bash
+docker buildx build . --platform linux/amd64,linux/arm64 --build-arg COMMIT=$(git rev-parse HEAD)
+```
+
+Speed improvements can be achieved during docker image build by relying on yarn [offline mirror](https://classic.yarnpkg.com/blog/2016/11/24/offline-mirror/).
+To configure this, run the following commands:
+
+```bash
+yarn config set yarn-offline-mirror .yarn/yarn-offline-mirror
+mv ~/.yarnrc ./ # `yarn config` commands are always global, make `.yarnrc` specific to this project
```

Comment:
What if someone has setting in their global yarn config? can't we just commit a .yarnrc to the project with this setting in it?

Reply:
If we do that, the offline mirror will always be enabled. Not sure if we want that by default.

Reply:
what's the disadvantage other than storage increase which is not a lot.
```
> du -sh .yarn
396M    .yarn
```
The advice to move the global yarn config into the current folder is not ideal, what if there are other settings in this file, means the global yarn config was effectively deleted.

If we wanna support offline mirror would just add a `.yarnrc` in git
```
yarn-offline-mirror ".yarn/yarn-offline-mirror"
yarn-offline-mirror-pruning true
```